storm-core/src/jvm/org/apache/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -219,9 +219,20 @@ public void setBlobMeta(String key, SettableBlobMeta meta, Subject who) throws A
     @Override
     public void deleteBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException {
         validateKey(key);
-        checkForBlobOrDownload(key);
-        SettableBlobMeta meta = getStoredBlobMeta(key);
-        _aclHandler.hasPermissions(meta.get_acl(), WRITE, who, key);
+
+        boolean checkPermission = true;
+        try {
+            checkForBlobOrDownload(key);
```

Comment:
Maybe we would need to check ZK and set checkPermission to false. 
If there's race condition so ZK node didn't exist but added later, we might skip pulling latest information and delete the blob in local anyway, but it should be occurred only from standby Nimbus and the blob will be re-pulled at next sync up.

I'll make appropriate change. Thanks.